File: backend/schemas/campaign.py
from datetime import datetime
from typing import Any

from pydantic import BaseModel, Field

from schemas.leads import Lead
from schemas.profile import FounderProfile


class CampaignCreate(BaseModel):
    name: str = Field(..., min_length=1, max_length=100)


class CampaignListItem(BaseModel):
    id: str
    name: str
    created_at: datetime
    updated_at: datetime


class CampaignFull(CampaignListItem):
    profile: FounderProfile | None = None
    whoami_extraction_id: str | None = None
    leads: list[Lead] = []


class CampaignProfileUpdate(BaseModel):
    profile: FounderProfile


class CampaignLeadsUpdate(BaseModel):
    leads: list[Any]  # Accept both Lead objects and legacy string format

File: frontend/src/types/campaign.ts
import type { FounderProfile } from './profile'

export interface CampaignListItem {
  id: string
  name: string
  created_at: string
  updated_at: string
}

export type LeadStatus = 'pending' | 'processing' | 'completed' | 'error'

export interface Lead {
  id: string
  query: string
  extraction_task_id: string | null
  verified_claims_id: string | null
  status: LeadStatus
  error: string | null
}

export interface Campaign extends CampaignListItem {
  profile: FounderProfile | null
  whoami_extraction_id: string | null
  leads: Lead[]
}
